src/main/java/com/hivemq/cli/commands/cli/PublishCommand.java

```diff
@@ -79,9 +75,12 @@ public PublishCommand(final @NotNull MqttClientExecutor mqttClientExecutor) {
     @CommandLine.Option(names = {"-q", "--qos"}, converter = MqttQosConverter.class, defaultValue = "0", description = "Quality of service for the corresponding topic (default for all: 0)", order = 1)
     @NotNull private MqttQos[] qos;
 
-    @CommandLine.Option(names = {"-m", "--message"}, converter = ByteBufferConverter.class, required = true, description = "The message to publish", order = 1)
+    @CommandLine.Option(names = {"-m", "--message"}, converter = ByteBufferConverter.class, description = "The message to publish", order = 1)
     @NotNull private ByteBuffer message;
 
+    @CommandLine.Option(names = {"-m:file", "--message-file"}, converter = FileToByteBufferConverter.class, description = "The message read in from a file", order = 1)
+    private void setMessageFromFile(final @NotNull ByteBuffer messageFromFile) { setMessage(messageFromFile); }
```

Comment:
I think it should be possible to define these as mutullay exclusive option via picocli (See https://picocli.info/#_mutually_exclusive_options) so that you don't need the null check below.

Reply:
Thanks, that definitively looks preferable!